### ISSUE TITLE
kitti dataset test config

### ISSUE THREAD

**Issue by grn213331** [YvanYin/Metric3D#217]
Hello, thank you very much for your excellent work. When testing on the KITTI dataset, I noticed that the errors I obtained are significantly higher than those reported in your paper. Could you please share your testing configuration? My current setup is listed below. Thank you!
`_base_=[
       '../_base_/models/encoder_decoder/dino_vit_small_reg.dpt_raft.py',
       '../_base_/datasets/_data_base_.py',
       '../_base_/default_runtime.py',
       ]
import numpy as np
model=dict(
    decode_head=dict(
        type='RAFTDepthNormalDPT5',
        iters=4,
        n_downsample=2,
        detach=False,
    )
)
max_value = 200
data_basic=dict(
    canonical_space = dict(
        img_size=(364, 1232),
        focal_length=1000,
    ),
    depth_range=(0, 1),
    depth_normalize=(0.1, 200),
    crop_size = (364, 1232),  # %28 = 0
    clip_depth_range=(0.1, 200),
    vit_size=(364,1232)
)
batchsize_per_gpu = 1
thread_per_gpu = 1`

**Comment by highlightz** [comment]
Hi @grn213331

Here are some conjectures to provide some ideas for your upcoming experiments. Before the authors or other professionals provide authoritative answers, these may offer some inspiration for the direction of your further testing.

These conjectures are derived from statements in the Metric3Dv2 paper and some logical guesses provided by Gemini.

As shown in the paper, if a Transformer architecture is used, the ViT backbone is implemented using DINO-v2:

<img width="1347" height="182" alt="Image" src="https://github.com/user-attachments/assets/39b84060-880d-4015-a1df-5725cefde0a7" />

------

(Gemini's conjecture) Generally speaking, ViT is sensitive to the input image size and aspect ratio (though DINO-v2 may have improved this to some extent).

As shown in the paper, during the training phase, in the canonical space, images are cropped to a resolution of 616 x 1064 for ViT training:
<img width="1235" height="81" alt="Image" src="https://github.com/user-attachments/assets/8ca42fda-4580-46f7-aa13-3cf6bf236a14" />

<img width="1361" height="173" alt="Image" src="https://github.com/user-attachments/assets/65135295-2b51-4c04-9fe1-9eb79f334bce" />

Official ViT configuration examples from this repository:
- mono/configs/HourglassDecoder/vit.raft5.small.py
- mono/configs/HourglassDecoder/vit.raft5.large.py
- mono/configs/HourglassDecoder/vit.raft5.giant2.py

All of these use a target resolution of `vit_size=(616,1064)`, which happens to be the resolution used during the training phase.
Note: These configurations are ultimately used in the `mono/utils/transform.py` script.

For long-strip images like those in the KITTI dataset (1242x375), the above processing method naturally results in a loss of horizontal field of view (FOV). For applications like blind-spot obstacle warnings in autonomous driving, this is generally unacceptable.

However, your configuration, `vit_size=(364,1232)`, has a resolution close to the original KITTI images. Its aspect ratio (approx. 3.38) differs significantly from the aspect ratio used during ViT training (approx. 1.73).

In practice, reaching this target resolution requires cropping in the vertical direction (see the aforementioned `transform.py`). For the scale accuracy of depth estimation, vertical feature density may be crucial (Gemini's conjecture), and vertical cropping directly leads to a reduction in vertical pixel density. Intuitively, in monocular depth estimation, the vertical position (v-coordinate) is a key geometric cue for the model to judge the distance of an object relative to the horizon. As seen in the images below, your configuration leads to the active discarding of a portion of pixels from the foreground ground (and similarly for the distant sky).

Furthermore, the inconsistency between inference and training resolutions may cause the results of positional encoding interpolation to deviate from the distribution of positional encodings during the training and fine-tuning phases, leading to an overall drop in accuracy.

In summary:

Dimension | Official Demo Config (1064×616) | Your Config (1232×364) | Conclusion
-- | -- | -- | --
Positional Encoding Stability | Extremely High (Matches training distribution) | Lower (Severe non-proportional interpolation) | Official is better: Affects global consistency
Vertical Feature Density | Rich (44 Patches) | Sparse (26 Patches) | Official is better: Important for ground plane perception and distant scene parsing
Geometric Deformation | None (Padding scheme) | None (Crop scheme) | No difference
Edge/Distant Accuracy | Stable | Prone to noise and fluctuations | Official is better
Effective FOV | Significant loss | Minimal loss | Yours is better


You can visually compare the images under the two target resolutions:

<img width="1383" height="1325" alt="Image" src="https://github.com/user-attachments/assets/a606b86b-15c5-40b2-9e0a-9e2a06e0399f" />
